packages/cosmos-client/src/CosmosClient.ts

```diff
@@ -0,0 +1,43 @@
+import { map } from "rxjs/operators";
+import { request, RequestResponse } from "@dcos/http-service";
+
+import Config from "#SRC/js/config/Config";
+import { buildRequestHeader, getErrorMessage } from "./utils";
+
+import { PackageVersionsResponse } from "../";
+
+export const CosmosClient = {
+  listPackageVersions: (packageName: string) =>
+    request<PackageVersionsResponse>(
+      `${Config.rootUrl || "/"}package/list-versions`,
+      {
+        method: "POST",
+        body: JSON.stringify({
+          includePackageVersions: true,
+          packageName
+        }),
+        headers: {
+          Accept: buildRequestHeader(
+            "list-versions",
+            "response",
+            "package",
+            "v1"
+          ),
+
+          "Content-type": buildRequestHeader(
+            "list-versions",
+            "request",
+            "package",
+            "v1"
+          )
+        }
+      }
+    ).pipe(
+      map((reqResp: RequestResponse<PackageVersionsResponse>) => {
```

Comment:
Let's embrace `partition` as the pattern for error handling. Semantically `map` is supposed to be doing mapping, not conditional errors. 

https://github.com/dcos/dcos-ui/pull/3579/files#diff-23616281663aac4135b3d9a84fb60f84R15

Reply:
Let's not embrace partition since partition is intended to be deprecated at some point reactivex/rxjs#3797

Lets rather solve it with filters: 

```js
const [success, error] = stream.pipe(map(source => 
  [
    source.pipe(filter(({code})=> code <= 200)), 
    source.pipe(filter(({code}) => code > 200))
  ]
))

```

This is basically what partition is doing and it is cleaner and can be used for multiple cases and not only binary cases.

Reply:
That's fair! Thanks for the research!

Reply:
Looks like a foundation/architecture-thingy to me.

Let's agree on a convention and then have this adapted afterwards.

Reply:
Let's please go with @Poltergeist suggestion merging two filtered streams

Reply:
Should we return two streams from the client and then merge them in the data-layer?

Filtering and merging in the client is a lot of code for what is accomplished with 4 lines in a map. I implemented in the ui-service client but immediately reverted bc it seemed wrong.

Reply:
we don't misuse `map` anymore but instead go with `tap`.